### PR TITLE
[enhancement](string-type) support create the table containing string/text type with a default value that length greater than 1

### LINKED ISSUE #12145: [Bug] create table failed when string type has a default value which length greater than 1
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

doris-master after doris 1.1.2

### What's Wrong?

create table failed when string type has a default value which length greater than 1

```

CREATE TABLE `tb6` (
  `id` int(11) NULL DEFAULT "10",
  `name` varchar(32) NULL DEFAULT "a varchar 32",
  `c_text` text NULL DEFAULT "a text",
  `c_str` string  NULL DEFAULT "a string"
) ENGINE=OLAP
UNIQUE KEY(`id`)
COMMENT 'OLAP'
DISTRIBUTED BY HASH(`id`) BUCKETS 4
PROPERTIES (
"replication_allocation" = "tag.location.default: 3",
"in_memory" = "false",
"storage_format" = "V2",
"disable_auto_compaction" = "false"
); 
```

### What You Expected?

create table successful

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by dujl**
support create the table containing string/text type with a default value that length greater than 1

# Proposed changes

Issue Number: close #12145

## Problem summary
Create the table containing string/text type with a default value that length greater than 1 failed.
```
CREATE TABLE `tb6` (
  `id` int(11) NULL DEFAULT "10",
  `name` varchar(32) NULL DEFAULT "a varchar 32",
  `c_text` text NULL DEFAULT "a text",
  `c_str` string  NULL DEFAULT "a string"
) ENGINE=OLAP
UNIQUE KEY(`id`)
COMMENT 'OLAP'
DISTRIBUTED BY HASH(`id`) BUCKETS 4
PROPERTIES (
"replication_allocation" = "tag.location.default: 3",
"in_memory" = "false",
"storage_format" = "V2",
"disable_auto_compaction" = "false"
); 
```

Describe your changes.
do not set string/text column type‘s length as 1.

```
mysql> insert into tb6 (id, c_str) select 2,  'string value';
Query OK, 1 row affected (0.04 sec)
{'label':'insert_6c0d9cb7483488e_9f029797c91dab0b', 'status':'VISIBLE', 'txnId':'2004'}
mysql> insert into tb6 (id, c_str) select 1,  'string value';
Query OK, 1 row affected (0.03 sec)
{'label':'insert_92da962f4f954d82_916b66273df28cea', 'status':'VISIBLE', 'txnId':'2005'}
mysql> 
mysql> select * from tb6;
+------+--------------+--------+--------------+
| id   | name         | c_text | c_str        |
+------+--------------+--------+--------------+
|    1 | a varchar 32 | a text | string value |
|    2 | a varchar 32 | a text | string value |
+------+--------------+--------+--------------+
2 rows in set (0.11 sec)
```

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [N ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ N] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ N] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ N] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java [hunk 1]
```diff
@@ -202,15 +202,6 @@ public void analyze(boolean isOlap) throws AnalysisException {
         }
         FeNameFormat.checkColumnName(name);
 
-        // When string type length is not assigned, it need to be assigned to 1.
-        if (typeDef.getType().isScalarType()) {
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java:206]: I think varchar can keep the old logic, only modify String logic.

**Comment by stalary** [comment]
Under normal circumstances should isAssignedStrLenInColDefinition to true, you are in what circumstances to this case?

**Comment by dujl** [comment]
> 

my case is create table with column 
```
`c_str` string  NULL DEFAULT "a string"
```
will failed.
FE has denied creation because of the default value length > 1

**Comment by stalary** [comment]
> 



> > 
> 
> my case is create table with column
> 
> ```
> `c_str` string  NULL DEFAULT "a string"
> ```
> > will failed. FE has denied creation because of the default value length > 1



> > 
> 
> my case is create table with column
> 
> ```
> `c_str` string  NULL DEFAULT "a string"
> ```
> 
> will failed. FE has denied creation because of the default value length > 1

Wait a minute, I try it in local.

**Review by stalary — APPROVED** [review]
LGTM

**Comment by stalary** [comment]
https://github.com/apache/doris/pull/12187 can resolve it.

**Pull request closed without merging**